### PR TITLE
CYS - Improve the error when a request fails due to permissions

### LINKED ISSUE #48657: [CYS] Users with the shop manager role fails to set up the foundations
### Prerequisites

- [X] I have carried out troubleshooting steps and I believe I have found a bug.
- [X] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

When a shop manager user clicks on the `Start designing` button, it fails to set up the foundations because the user does not have the required permissions.

Screenshot 2024-06-20 at 10 57 35

However, when users go back to the task list, they can see the task as completed.

https://github.com/woocommerce/woocommerce/assets/4344253/9171d70e-87fd-4381-9970-b8656579df48

### Expected behavior

Ideally, the shop manager user should be able to complete the CYS task. However, the task requires users to have the install theme permission so it makes sense that the user cannot complete the task.

I don't know what the best way to handle this is. Maybe we can show a better error message when they don't have the required permissions. Besides, the task should not be marked as completed if the user does not have the required permissions.

### Actual behavior

The user with the shop manager role can see the task as completed and when they revisit the task, they can see the `Customize your theme` button.

### Steps to reproduce

1. Create a fresh site
2. Complete or skip the core profiler
3. Create a new user with shop manger role
4. Login with the new user
5. Go to the CYS task
6. Click on `Start designing` button

### WordPress Environment

`
### WordPress Environment ###

WordPress address (URL): https://usually-liable.jurassic.ninja
Site address (URL): https://usually-liable.jurassic.ninja
WC Version: 9.1.0
Legacy REST API Package Version: The Legacy REST API plugin is not installed on this site.
Action Scheduler Version: ✔ 3.8.0
Log Directory Writable: ✔
WP Version: 6.5.4
WP Multisite: –
WP Memory Limit: 512 MB
WP Debug Mode: ✔
WP Cron: ✔
Language: en_US
External object cache: ✔

### Server Environment ###

Server Info: nginx
PHP Version: 8.1.29
PHP Post Max Size: 2 GB
PHP Time Limit: 1200
PHP Max Input Vars: 6144
cURL Version: 8.7.1
OpenSSL/1.1.1w

SUHOSIN Installed: –
MySQL Version: 10.6.15-MariaDB-log
Max Upload Size: 2 GB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 9.1.0
WC Database Prefix: wp_
Total Database Size: 8.47MB
Database Data Size: 5.19MB
Database Index Size: 3.28MB
wp_woocommerce_sessions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_api_keys: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_attribute_taxonomies: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_downloadable_product_permissions: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_order_items: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_order_itemmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_tax_rates: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_tax_rate_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zones: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_shipping_zone_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zone_methods: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_payment_tokens: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_payment_tokenmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_log: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_actions: Data: 0.02MB + Index: 0.11MB + Engine InnoDB
wp_actionscheduler_claims: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_groups: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_logs: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_commentmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_comments: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_gla_attribute_mapping_rules: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_gla_budget_recommendations: Data: 0.19MB + Index: 0.14MB + Engine InnoDB
wp_gla_merchant_issues: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_gla_shipping_rates: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_gla_shipping_times: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_jetpack_sync_queue: Data: 0.06MB + Index: 0.06MB + Engine InnoDB
wp_links: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_mailpoet_automations: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_mailpoet_automation_runs: Data: 0.02MB + Index: 0.08MB + Engine InnoDB
wp_mailpoet_automation_run_logs: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_mailpoet_automation_run_subjects: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_mailpoet_automation_triggers: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_mailpoet_automation_versions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_mailpoet_custom_fields: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_mailpoet_dynamic_segment_filters: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_mailpoet_feature_flags: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_mailpoet_forms: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_mailpoet_log: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_mailpoet_migrations: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_mailpoet_newsletters: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_mailpoet_newsletter_links: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_mailpoet_newsletter_option: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_mailpoet_newsletter_option_fields: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_mailpoet_newsletter_posts: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_mailpoet_newsletter_segment: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_mailpoet_newsletter_templates: Data: 2.52MB + Index: 0.00MB + Engine InnoDB
wp_mailpoet_scheduled_tasks: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_mailpoet_scheduled_task_subscribers: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_mailpoet_segments: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_mailpoet_sending_queues: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_mailpoet_settings: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_mailpoet_statistics_bounces: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_mailpoet_statistics_clicks: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_mailpoet_statistics_forms: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_mailpoet_statistics_newsletters: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_mailpoet_statistics_opens: Data: 0.02MB + Index: 0.08MB + Engine InnoDB
wp_mailpoet_statistics_unsubscribes: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_mailpoet_statistics_woocommerce_purchases: Data: 0.02MB + Index: 0.08MB + Engine InnoDB
wp_mailpoet_stats_notifications: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_mailpoet_subscribers: Data: 0.02MB + Index: 0.23MB + Engine InnoDB
wp_mailpoet_subscriber_custom_field: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_mailpoet_subscriber_ips: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_mailpoet_subscriber_segment: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_mailpoet_subscriber_tag: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_mailpoet_tags: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_mailpoet_user_agents: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_mailpoet_user_flags: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_options: Data: 0.34MB + Index: 0.06MB + Engine InnoDB
wp_postmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_posts: Data: 0.11MB + Index: 0.06MB + Engine InnoDB
wp_termmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_terms: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_term_relationships: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_term_taxonomy: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_usermeta: Data: 0.09MB + Index: 0.03MB + Engine InnoDB
wp_users: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_admin_notes: Data: 0.06MB + Index: 0.00MB + Engine InnoDB
wp_wc_admin_note_actions: Data: 0.06MB + Index: 0.02MB + Engine InnoDB
wp_wc_category_lookup: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_customer_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_download_log: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_orders: Data: 0.02MB + Index: 0.11MB + Engine InnoDB
wp_wc_orders_meta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_addresses: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_wc_order_coupon_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_operational_data: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_product_lookup: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_wc_order_stats: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_order_tax_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_product_attributes_lookup: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_product_download_directories: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_product_meta_lookup: Data: 0.02MB + Index: 0.11MB + Engine InnoDB
wp_wc_rate_limits: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_reserved_stock: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_tax_rate_classes: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_webhooks: Data: 0.02MB + Index: 0.02MB + Engine InnoDB

### Post Type Counts ###

attachment: 1
mailpoet_page: 1
page: 7
post: 3
revision: 1
wp_font_face: 33
wp_font_family: 12
wp_global_styles: 1
wp_navigation: 1
wp_template: 1

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ✔

### Active Plugins (10) ###

Akismet Anti-spam: Spam Protection: by Automattic - Anti-spam Team – 5.3.2
Google Listings and Ads: by WooCommerce – 2.7.3
Jetpack: by Automattic – 13.6-a.5
MailPoet: by MailPoet – 4.52.0
Pinterest for WooCommerce: by WooCommerce – 1.3.24
TikTok: by TikTok – 1.2.6
WooCommerce: by Automattic – 9.1.0-dev-9588915199-g0a411dd
WooCommerce Beta Tester: by WooCommerce – 2.3.1
WooPayments: by Automattic – 7.8.0
WooCommerce Shipping & Tax: by WooCommerce – 2.6.0

### Dropin Plugins () ###

advanced-cache.php: advanced-cache.php
object-cache.php: Memcached

### Must Use Plugins (1) ###

load.php: by  –

### Settings ###

API Enabled: –
Force SSL: –
Currency: GBP (£)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to WooCommerce.com: –
Enforce Approved Product Download Directories: ✔
HPOS feature enabled: ✔
Order datastore: Automattic\WooCommerce\Internal\DataStores\Orders\OrdersTableDataStore
HPOS data sync enabled: –

### Logging ###

Enabled: ✔
Handler: Automattic\WooCommerce\Internal\Admin\Logging\LogHandlerFileV2
Retention period: 30 days
Level threshold: –
Log directory size: 34 KB

### WC Pages ###

Shop base: #5 - /shop/
Cart: #6 - /cart/ -  Contains the woocommerce/cart block
Checkout: #7 - /checkout/ -  Contains the woocommerce/checkout block
My account: #8 - /my-account/
Terms and conditions: ❌ Page not set

### Theme ###

Name: Twenty Twenty-Four
Version: 1.1
Author URL: https://wordpress.org
Child Theme: ❌ – If you are modifying WooCommerce on a parent theme that you did not build personally we recommend using a child theme. See: How to create a child theme
WooCommerce Support: ❌ Not declared

### Templates ###

Overrides: /srv/htdocs/wp-content/plugins/wc_beta_tester_live_branch_9.1.0-dev-9588915199-g0a411dd/templates/block-notices/error.php
/srv/htdocs/wp-content/plugins/wc_beta_tester_live_branch_9.1.0-dev-9588915199-g0a411dd/templates/block-notices/notice.php
/srv/htdocs/wp-content/plugins/wc_beta_tester_live_branch_9.1.0-dev-9588915199-g0a411dd/templates/block-notices/success.php

### WooPayments ###

Version: 7.8.0
Connected to WPCOM: Yes
WPCOM Blog ID: 234374948
Account ID: ❌ Not connected
Logging: Disabled

### MailPoet ###

Sending Method: PHPMail
Send all site's emails with: Default WordPress sending method
Task Scheduler method: Action Scheduler
Cron ping URL: https://usually-liable.jurassic.ninja?mailpoet_router&endpoint=cron_daemon&action=ping

### Admin ###

Enabled Features: activity-panels
analytics
product-block-editor
coupons
core-profiler
customize-store
customer-effort-score-tracks
import-products-task
experimental-fashion-sample-products
shipping-smart-defaults
shipping-setting-tour
homescreen
marketing
mobile-app-banner
navigation
onboarding
onboarding-tasks
product-custom-fields
remote-inbox-notifications
remote-free-extensions
payment-gateway-suggestions
shipping-label-banner
subscriptions
store-alerts
transient-notices
woo-mobile-welcome
wc-pay-promotion
wc-pay-welcome-page
launch-your-store

Disabled Features: experimental-blocks
minified-js
pattern-toolkit-full-composability
product-pre-publish-modal
settings
async-product-editor-category-field
product-editor-template-system
beta-tester-slotfill-examples

Daily Cron: ✔ Next scheduled: 2024-06-21 02:08:55 +00:00
Options: ✔
Notes: 60
Onboarding: completed

### Action Scheduler ###

Complete: 28
Oldest: 2024-06-20 02:10:25 +0000
Newest: 2024-06-20 03:11:11 +0000

Failed: 1
Oldest: 2024-06-20 02:11:33 +0000
Newest: 2024-06-20 02:11:33 +0000

Pending: 5
Oldest: 2024-06-20 03:13:11 +0000
Newest: 2024-06-21 02:47:11 +0000

### Status report information ###

Generated at: 2024-06-20 03:13:30 +00:00
`

### Isolating the problem

- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/products/storefront/).
- [X] I can reproduce this bug consistently using the steps above.

### PR DESCRIPTION
**Pull request by albarin**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Closes https://github.com/woocommerce/woocommerce/issues/48657

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

0. Go to `wp-admin/tools.php?page=woocommerce-admin-test-helper`, and run the `Reset Customize Your Store` command on the `Tools` tab.
1. Create a new user with `Shop manager` role.
2. Login with the new user.
3. Go to `wp-admin/admin.php?page=wc-admin&path=%2Fcustomize-store` and click `Start designing`.
4. Click on `Start designing` button.
5. Make sure you are redirected again to the `Intro` screen with an error about permissions (instead of the generic error one).

### Changelog entry

-   [x] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [ ] Patch
-   [x] Minor
-   [ ] Major

#### Type

-   [x] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 
CYS - Improve the error when a request fails due to permissions
</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

**Review by gigitux — CHANGES_REQUESTED** [review]
Thanks for working on this, Alba! 

I see the right label when I start the flow with a user with Shop manager role:


https://github.com/user-attachments/assets/b3066daa-04d7-42b4-ac42-7f1d5f94287f





At the same time, it looks like that when there is an error on another endpoint, the flow isn't stopped. Via dev tools I blocked the `/products` endpoint (necessary to create the products):










| Before | After |
|--------|--------|
| <video src="https://github.com/user-attachments/assets/6d217067-9143-4200-9a31-35570b44444c" /> | <video src="https://github.com/user-attachments/assets/73807b16-da2d-4233-a86c-0b6fbd8292a5" /> |


Checking the changes, I don't understand why there is this regression :/

**Comment by albarin** [comment]
Thanks for the review @gigitux, I think this fixes it https://github.com/woocommerce/woocommerce/pull/50211/commits/62419d31107dd5a3dc86d92873c7442bfca8924d also updates the copy according to Laura suggestion.

File: plugins/woocommerce-admin/client/customize-store/design-without-ai/actions.ts [hunk 1]
```diff
@@ -27,8 +27,15 @@ const redirectToIntroWithError = sendParent<
 	DesignWithoutAIStateMachineContext,
 	DesignWithoutAIStateMachineEvents,
 	DesignWithoutAIStateMachineEvents
->( {
-	type: 'NO_AI_FLOW_ERROR',
+>( ( context, event ) => {
+	const errorEvent = event as {
+		type: string;
+		data: { data: { status: number } };
```
Comment (gigitux) [inline plugins/woocommerce-admin/client/customize-store/design-without-ai/actions.ts]: We can improve the type here:
```suggestion
		data?: { data?: { status: number } };
```

File: plugins/woocommerce-admin/client/customize-store/design-without-ai/actions.ts [hunk 2]
```diff
@@ -27,8 +27,15 @@ const redirectToIntroWithError = sendParent<
...
+	};
+	return {
+		type: 'NO_AI_FLOW_ERROR',
+		errorStatus: errorEvent?.data?.data?.status || undefined,
```
Comment (gigitux) [inline plugins/woocommerce-admin/client/customize-store/design-without-ai/actions.ts]: `|| undefined` isn't necessary
```suggestion
		errorStatus: errorEvent?.data?.data?.status
```

File: plugins/woocommerce-admin/client/customize-store/intro/actions.ts [hunk 3]
```diff
@@ -130,8 +130,13 @@ export const assignNoAIFlowError = assign<
 	customizeStoreStateMachineContext,
 	customizeStoreStateMachineEvents
 >( {
-	intro: ( context ) => {
-		return { ...context.intro, hasErrors: true };
+	intro: ( context, event: unknown ) => {
+		const { errorStatus } = event as { errorStatus: number };
```
Comment (gigitux) [inline plugins/woocommerce-admin/client/customize-store/intro/actions.ts]:
```suggestion
		const { errorStatus } = event as { errorStatus: number | undefined };
```

**Review by gigitux — APPROVED** [review]
Thanks for working on this! LGTM! I left some minor comments about types that it could make sense to address before merging the PR! 🚢

**Pull request merged**